sdk/core/core-rest-pipeline/src/util/file.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { isReadableStream } from "./typeGuards";
+import { toWebStream } from "./stream";
+
+/**
+ * Options passed into createFile specifying metadata about the file.
+ */
+export interface CreateFileOptions {
+  /**
+   * The MIME type of the file.
+   */
+  type?: string;
+
+  /**
+   * Last modified time of the file as a UNIX timestamp.
+   * This will default to the current date.
+   */
+  lastModified?: number;
+
+  /**
+   * relative path of this file when uploading a directory.
+   */
+  webkitRelativePath?: string;
+}
+
+/**
+ * Extra options for createFile when a stream is being passed in.
+ */
+export interface CreateFileFromStreamOptions extends CreateFileOptions {
+  /**
+   * Size of the file represented by the stream in bytes.
+   *
+   * This will be used by the pipeline when calculating the Content-Length header
+   * for the overall request.
+   */
+  size?: number;
+}
+
+/**
+ * Create an object that implements the File interface. This object is intended to be
+ * passed into RequestBodyType.formData, and is not guaranteed to work as expected in
+ * other situations.
+ *
+ * Use this function to:
+ * - Create a File object for use in RequestBodyType.formData in environments where the
+ *   global File object is unavailable.
+ * - Create a File-like object from a readable stream without reading the stream into memory.
+ *
+ * @param content - the content of the file as a stream. When a File object made using createFile is
+ *                  passed in a request's form data map, the stream will not be read into memory
+ *                  and instead will be streamed when the request is made.
+ * @param name - the name of the file.
+ * @param options - optional metadata about the file, e.g. file name, file size, MIME type.
+ */
+export function createFile(
```

Comment:
Did we consider moving this to core-util instead? It doesn't seem to be related to concepts normally exported by core-rest-pipeline.

Reply:
This fill is specifically for `FormDataPolicy` and may not work as expected in other contexts. I'd rather it stay close to where it's intended to be used.

Reply:
Yeah for now but the general concept is creating a file from a stream which may be useful outside the form data policy context. For instance could be handy for requests that contain just a file stream and not a of form data content type. I think the name should reflect the streamability part.

Reply:
I'm ok with starting it here and we can always re-export it from core-util if that made more sense in the future, though I'm not sure that core-amqp would ever need something like this